client_plugin/drivers/shared/etcdops.go

```diff
@@ -0,0 +1,271 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	etcdClient "github.com/coreos/etcd/clientv3"
+	"github.com/docker/engine-api/client"
+	"github.com/docker/engine-api/types"
+	"github.com/docker/engine-api/types/swarm"
+)
+
+const (
+	apiVersion               = "v1.24" // docker engine 1.24 and above support this api version
+	dockerUSocket            = "unix:///var/run/docker.sock"
+	etcdClientPort           = ":2379"
+	etcdPeerPort             = ":2380"
+	etcdClusterToken         = "vsphere-shared-etcd-cluster"
+	etcdListenURL            = "0.0.0.0"
+	etcdScheme               = "http://"
+	etcdClusterStateNew      = "new"
+	etcdClusterStateExisting = "existing"
+)
+
+// InitEtcd start or join ETCD cluster depending on the role of the node
+func InitEtcd() error {
+	// create new docker client
```

Comment:
I was adding similar things to shared_plugin.go. Should we move the docker Client to VolumeDriver structure and the docker info part to a function in shared_plugin.go?

We might want to call docker info multiple times at various places. Perhaps a better place will be shared_plugin.go. A function getsDockerInfo() can return a structure similar to this:

53 // Docker swarm information
 54 type SwarmMetadata struct {
 55         isManager bool
 56         isLeader  bool
 57         selfIP    string
 58         managers  []string
 59 }